### PR TITLE
[doc] Move documentation to readthedocs.org

### LINKED ISSUE #688: Versions v2.13-v2.16 not found in documentation
When selecting another version from the latest documentation page, versions v2.13-v2.16 throw a `Page not found` error.

### PR DESCRIPTION
**Pull request by kraushm**
Modifications based on moving the documentation to [readthedocs.org](reframe-hpc.readthedocs.org)

Closes #688

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -112,7 +112,7 @@
     'collapse_navigation': True,
     'display_version': True,
     'navigation_depth': 5,
-    'analytics_id': 'UA-86976416-6',
+    # 'analytics_id': 'UA-86976416-6',
```
Comment (vkarak) [inline docs/conf.py]: Remove this completely from here.

**Comment by kraushm** [comment]
@jenkins-cscs retry daint

File: README.md [hunk 2]
```diff
@@ -16,12 +16,12 @@ Users can create their own test hierarchies, create test factories for generatin
 
 ## Documentation
 
-The official documentation is maintained [here](https://eth-cscs.github.io/reframe/index.html).
+The official documentation is maintained [here](https://reframe-hpc.readthedocs.org).
```
Comment (vkarak) [inline README.md]: If the redirection from the old page succeeds, we don't need to change the README here.

File: README.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-[![Build Status](https://travis-ci.org/eth-cscs/reframe.svg?branch=master)](https://travis-ci.org/eth-cscs/reframe) [![codecov.io](https://codecov.io/gh/eth-cscs/reframe/branch/master/graph/badge.svg)](https://codecov.io/github/eth-cscs/reframe)
+[![Build Status](https://travis-ci.org/eth-cscs/reframe.svg?branch=master)](https://travis-ci.org/eth-cscs/reframe) [![codecov.io](https://codecov.io/gh/eth-cscs/reframe/branch/master/graph/badge.svg)](https://codecov.io/github/eth-cscs/reframe) [![Documentation Status](https://readthedocs.org/projects/reframe-hpc/badge/?version=stable)](https://reframe-hpc.readthedocs.io/en/stable/?badge=stable)
```
Comment (vkarak) [inline README.md]: You should use the `latest` here. These badges always reflect the current status of master.

**Pull request merged**